test/variable.c

```diff
@@ -41,7 +41,7 @@ const char* (*const volatile function_pointer_with_multiple_qualifiers)(const ch
 /**
  * a complex type involving function pointers somehow
  */
-const char* (*const *volatile *restrict legal_type_involving_function_pointer[12][3])(const char *in);
+const char* (*const *volatile *const legal_type_involving_function_pointer[12][3])(const char *in);
```

Comment:
'restrict' is a case in `_type_fixup()` so I'd keep that, but if necessary split it to a separate test.

Reply:
True, I thought we tested the `restrict` path in other tests already, which is not the case apparently.

My suggestion would be to keep the change here, add other cases for various forms of `restrict` that compile in C++ as well and have a smaller test case for the C domain only with this particular case if you think the combination of features is relevant. Though I don't understand what crucial aspect makes it C specific so I don't know what to call it...

By the way, this prompted me to check the standards and I still don't understand the difference between C and C++. I'd need to change the commit message at least (for no discernable reason, I thought `restrict` was limited to function arguments and was wrongly questioning the validity of the snippet). Do you have any guesses as to why it won't work in C++?

FYI, this is what I get:
```
bruno@bms-tp ~ $ g++ a.cpp 
a.cpp:1:40: error: expected ‘)’ before ‘legal_type_involving_function_pointer’
    1 | const char* (*const *volatile *restrict legal_type_involving_function_pointer[12][3])(const char *in);
      |             ~                          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      |                                        )

bruno@bms-tp ~ $ clang++ a.cpp 
a.cpp:1:41: error: expected ')'
const char* (*const *volatile *restrict legal_type_involving_function_pointer[12][3])(const char *in);
                                        ^
a.cpp:1:13: note: to match this '('
const char* (*const *volatile *restrict legal_type_involving_function_pointer[12][3])(const char *in);
            ^
1 error generated.
```

Reply:
So, on a 2nd look, `restrict` does not seem to be part of C++ at all, but both gcc and clang accept them in some cases (no-op or not I couldn't say). Even this will not work in C++:
```c
int foo(int *restrict bar);
```

This only makes the particular case here weirder though, because this works in C++ and is conceptually the same albeit a much simpler version of the snippet that triggered this commit:
```c
int *restrict foo;
```

Anyway, I think I have an acceptable approach by putting all `restrict` stuff in its own test unit.